protocol/nats_jetstream/v3/sender.go

```diff
@@ -0,0 +1,139 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+)
+
+type sender struct {
+	// nats connection
+	conn      *nats.Conn
+	url       string
+	natsOpts  []nats.Option
+	connOwned bool
+
+	// jetstream options and configuration
+	jetSteamOpts []jetstream.JetStreamOpt
+	jetStream    jetstream.JetStream
+	publishOpts  []jetstream.PublishOpt
+
+	subject string
+}
+
+// NewSender creates a new protocol.Sender responsible for opening and closing the NATS connection
+// Use WithURL() OR WithConnection()
+func NewSender(ctx context.Context, subject string, opts ...ProtocolOption) (*sender, error) {
```

Comment:
Can we make these sender/receiver constructors private?

Reply:
The protocol constructor makes both a sender and receiver.  in some cases you might only want a sender OR receiver.